iib/web/models.py

```diff
@@ -1433,3 +1434,110 @@ def validate_registry_auths(registry_auths):
+
+        :param dict labels: the dictionary of the labels or ``None``
+        """
+        self._labels = json.dumps(labels, sort_keys=True) if labels is not None else None
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the create-empty-index API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+        if request_kwargs.get('from_index') is None:
+            raise ValidationError('"from_index" must be a specified')
+        if (
+            not isinstance(request_kwargs.get('from_index'), str)
+            or len(request_kwargs.get('from_index')) == 0
+        ):
+            raise ValidationError('"from_index" must be a non-empty string')
+
+        new_labels = request_kwargs.get('labels')
+        if new_labels is not None:
+            if not isinstance(new_labels, dict):
+                raise ValidationError('The value of "labels" must be a JSON object')
+
+            for key, value in new_labels.items():
+                if not isinstance(value, str) or not isinstance(key, str):
+                    raise ValidationError(f'The key and value of "{key}" must be a string')
+
+        for arg in ('add_arches', 'overwrite_from_index', 'overwrite_from_index_token'):
+            if arg in request_kwargs:
+                raise ValidationError(
+                    f'The "{arg}" arg is invalid for the create-empty-index endpoint.'
+                )
+
+        cls._from_json(
+            request_kwargs,
+            additional_required_params=['from_index'],
+            additional_optional_params=['binary_image', 'labels'],
+            batch=batch,
+        )
+
+        request = cls(**request_kwargs)
+        request.add_state('in_progress', 'The request was initiated')
+
+        return request
+
+    def to_json(self, verbose=True):
+        """
+        Provide the JSON representation of an "create-empty-index" build request.
+
+        :param bool verbose: determines if the JSON output should be verbose
+        :return: a dictionary representing the JSON of the build request
+        :rtype: dict
+        """
+        rv = super().to_json(verbose=verbose)
+        rv.update(self.get_common_index_image_json())
+        rv.pop('bundles')
+        rv.pop('bundle_mapping')
+        rv.pop('organization')
+        rv.pop('deprecation_list')
+        rv.pop('removed_operators')
+        rv['labels'] = self.labels
+
+        return rv
+
+    def get_mutable_keys(self):
+        """
+        Return the set of keys representing the attributes that can be modified.
+
+        :return: a set of key names
+        :rtype: set
+        """
+        rv = super().get_mutable_keys()
+        rv.update(self.get_index_image_mutable_keys())
+        rv.update('labels')
+
+        rv.remove('from_bundle_image_resolved')
+
```

Comment:
Could you please remove one of the blank lines?

Reply:
sure